ProcessMaker/processmaker
# Implement Screen Template Ellipsis Menu Permission: Publish Screen Template

This PR introduces the new permission, "Publish Screen Templates," to the My Templates and Public Templates ellipsis menu.

## Changes Made

- Modified the Screen Template ellipsis menu to include the "Make Public" action for templates owned by users with the "Publish Screen Templates" permission.
- Restricted the "Edit" and "Delete" actions in the ellipsis menu for public templates to users with the appropriate permissions.

## How to Test

1. Create a user without the "Publish Screen Templates" permission.
2. Log in as the user and navigate to Designer > Screens > My Templates.
3. Verify that the ellipsis menu for each template does not include the "Make Public" action.
4. Navigate to Public Templates.
5. Ensure that the ellipsis menu for each template does not display the "Edit" and "Delete" actions.
6. Enable the "Publish Screen Templates" permission for the user.
7. Repeat steps 2-5 and verify that the "Make Public" action is now available in My Templates and the "Edit" and "Delete" actions are available in Public Templates.


Note: Functionality to execute actions (e.g., making a template public) is not yet implemented in this PR and will need to be tested separately once available.

## Related Tickets & Packages
- [FOUR-13485](https://processmaker.atlassian.net/browse/FOUR-13485)

ci:next

## Code Review Checklist


[FOUR-13485]: https://processmaker.atlassian.net/browse/FOUR-13485?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
